packages/client-core/src/user/UserUISystem.tsx

```diff
@@ -27,28 +27,42 @@ import React, { useEffect } from 'react'
 
 import { defineSystem } from '@ir-engine/ecs/src/SystemFunctions'
 import { PresentationSystemGroup } from '@ir-engine/ecs/src/SystemGroups'
-import { getMutableState } from '@ir-engine/hyperflux'
+import { getMutableState, getState } from '@ir-engine/hyperflux'
 
 import { useHookstate } from '@hookstate/core'
 import useFeatureFlags from '@ir-engine/client-core/src/hooks/useFeatureFlags'
 import { FeatureFlags } from '@ir-engine/common/src/constants/FeatureFlags'
 import { EngineState, QueryReactor, useEntityContext, useOptionalComponent } from '@ir-engine/ecs'
-import { IFrameComponent } from '@ir-engine/engine/src/scene/components/IFrameComponent'
+import { PopoverComponent, PopoverComponentState } from '@ir-engine/engine/src/scene/components/PopoverComponent'
 import { NetworkState } from '@ir-engine/network'
 import { PopoverState } from '../common/services/PopoverState'
 import { InviteService } from '../social/services/InviteService'
 import { ViewerMenuState } from '../util/ViewerMenuState'
 import EmbedFrame from './menus/avatar/EmbedFrame'
 
-const IFrameReactor = () => {
+const PopoverReactor = () => {
   const entity = useEntityContext()
-  const iframeComponent = useOptionalComponent(entity, IFrameComponent)
+  const popoverComponent = useOptionalComponent(entity, PopoverComponent)
+  const popoverComponentState = getMutableState(PopoverComponentState)
 
   useEffect(() => {
-    if (iframeComponent?.isOpen.value) {
-      PopoverState.showPopupover(<EmbedFrame src={iframeComponent?.src.value} />)
+    popoverComponentState.merge({
+      iframe: EmbedFrame
+    })
+  }, [])
+
+  useEffect(() => {
+    if (popoverComponent?.isOpen.value) {
+      const popoverType = popoverComponent?.type.value || 'iframe'
+      const Component = getState(PopoverComponentState)[popoverType]
+      PopoverState.showPopupover(
+        // todo: add more animation type and save it in ecs
+        <div className="animate-slideIn">
+          <Component src={popoverComponent?.src.value} />
+        </div>
+      )
     }
-  }, [iframeComponent])
+  }, [popoverComponentState, popoverComponent])
```

Comment:
the dependencies don't make too much sense here. `!!popoverComponent?.isOpen.value` should be sufficient to identify whether it has gone from not open to open, and that should be the only time this effect triggers